lib/policies/load-balancing.js

```diff
@@ -319,39 +336,32 @@ TokenAwareIterator.prototype.iterate = function (callback) {
 };
 
 TokenAwareIterator.prototype.computeNext = function () {
-  //return local replicas
-  var host = this.nextLocalReplica();
-  if (host) {
-    return {value: host, done: !host};
+  var host;
+  if (this.localReplicaIndex < this.localReplicas.length) {
+    host = this.localReplicas[(this.startIndex + (this.localReplicaIndex++)) % this.localReplicas.length];
+    return { value: host, done: false };
   }
-  //return child hosts
-  var item = this.childIterator.next();
-  if (!item.done) {
+  this.startIndex = -1;
+  // Return hosts from child policy
+  var item;
+  while ((item = this.childIterator.next()) && !item.done) {
     if (this.replicaMap[item.value.address]) {
-      return this.computeNext();
+      // Avoid yielding replicas from the child load balancing policy query plan
```

Comment:
With 2 now including remote replicas if emitted by the child policy, is 3 needed?  The only case where it would emit something is where the child policy doesn't contain the remote replicas, and if it doesn't there is probably a reason it doesn't (although you could use that same line of reasoning for the local replicas).